File: 2025/day5/main.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

type Range struct {
	start int
	end   int
}

func main() {
	data, _ := os.ReadFile("input.txt")
	content := string(data)

	// Split by blank line
	parts := strings.Split(strings.TrimSpace(content), "\n\n")
	rangeLines := strings.Split(parts[0], "\n")
	idLines := strings.Split(parts[1], "\n")

	// Parse ranges
	var ranges []Range
	for _, line := range rangeLines {
		parts := strings.Split(line, "-")
		start, _ := strconv.Atoi(parts[0])
		end, _ := strconv.Atoi(parts[1])
		ranges = append(ranges, Range{start, end})
	}

	// Parse available IDs
	var availableIDs []int
	for _, line := range idLines {
		id, _ := strconv.Atoi(line)
		availableIDs = append(availableIDs, id)
	}

	// Count fresh IDs
	freshCount := 0
	for _, id := range availableIDs {
		isFresh := false
		for _, r := range ranges {
			if id >= r.start && id <= r.end {
				isFresh = true
				break
			}
		}
		if isFresh {
			freshCount++
		}
	}

	fmt.Println(freshCount)
}

File: 2025/day5/main.py
with open("input.txt", "r") as f:
    content = f.read()

# Split by blank line
parts = content.strip().split("\n\n")
range_lines = parts[0].split("\n")
id_lines = parts[1].split("\n")

# Parse ranges
ranges = []
for line in range_lines:
    start, end = map(int, line.split("-"))
    ranges.append((start, end))

# Parse available IDs
available_ids = [int(line) for line in id_lines]

# Count fresh IDs
fresh_count = 0
for id_val in available_ids:
    is_fresh = False
    for start, end in ranges:
        if start <= id_val <= end:
            is_fresh = True
            break
    if is_fresh:
        fresh_count += 1

print(fresh_count)
